### ISSUE TITLE
Postcss and autoprefixer peer dependencies

### ISSUE THREAD

**Issue by bramstein** [gustavnikolaj/express-autoprefixer#11]
Why are `postcss` and `autoprefixer` peer dependencies instead of dependencies? It looks like the code depends on them being installed, so perhaps they should be dependencies?

**Comment by gustavnikolaj** [comment]
When you use express-autoprefixer, you are using something else to do the autoprefixing in your production builds, in my case assetgraph. Part of the autoprefixer package is a snapshot of some caniuse.com data, that it uses to decide what to prefix (in conjunction with your options). Having different data deciding what to prefix in development and production can lead to surprising differences that are really hard to catch.

It might be too far fetched - I don't know - but that's the reasoning at least.

Things have probably a thousand times since we wrote this module, and with npm 3 changing how it does peer dependencies, we might want to reevaluate. What is the problem that you're encountering?

**Comment by bramstein** [comment]
I'm using npm 3 and no matter what I do it keeps complaining about unmet peer dependencies. I'm using this module as middleware for a custom (development) server.

**Comment by gustavnikolaj** [comment]
npm will from version 3 no longer automatically install peer dependencies. so you need to install autoprefixer in addition to express-autoprefixer.

I'm ready to reevaluate the decision, but I just need to understand the problem that you're having.

**Comment by bramstein** [comment]
If you have npm 3 installed and do:

```
npm install autoprefixer postcss express-autoprefixer
npm ls
```

It'll keep complaining about unmet peer dependencies (even though the dependencies are installed).

**Comment by gustavnikolaj** [comment]
```
npm WARN express-autoprefixer@4.0.3 requires a peer of autoprefixer@5 but none was installed.
npm WARN express-autoprefixer@4.0.3 requires a peer of postcss@4 but none was installed.
```

When installing the packages like that you get the latest version. The latest version of postcss is 5.0.16 and the latest version of autoprefixer is 6.3.3. None of the latest versions match the peer dep requirements.

Running the following should work:

```
$ npm install autoprefixer@5 express-autoprefixer postcss@4
```

There's a PR for updating the dependencies in #10 that I'll take a look at at the soonest.

**Comment by gustavnikolaj** [comment]
Would you prefer the ease of an permissive semver dependency when working with npm 3?

I haven't made the upgrade still, so I have no experience working with the new peer dependency system. This might be too inconvenient.

**Comment by bramstein** [comment]
Thanks, that fixed the issue. It's a little confusing, perhaps better messaging from npm would help.

**Comment by gustavnikolaj** [comment]
I added liberal dependencies for autoprefixer and postcss as well as peerdependencies. It seems to work from my testing. If you upgrade to version 4.0.6 it should work without having to manually install other things, even with npm 3.

I also released a new major version (5.0.0) that uses newer versions of both autoprefixer and postcss. But the never versions no longer support node 0.10.

I hope that this will fix the issue, and sorry for the inconvenience.
